packages/terra-form-select/src/native-select/NativeSelect.jsx

```diff
@@ -0,0 +1,247 @@
+  id: PropTypes.string.isRequired,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: intlShape.isRequired,
+  /**
+   * Whether the input is invalid.
+   */
+  isInvalid: PropTypes.bool,
+  /**
+   * Whether the field displays as Incomplete. Use when no value has been provided. _(usage note: `required` must also be set)_.
+   */
+  isIncomplete: PropTypes.bool,
+  /**
+   * Callback function triggered when the select value changes. function(value)
+   */
+  onChange: PropTypes.func,
+  /**
+   * The array of select options and opt groups.
+   */
+  options: PropTypes.arrayOf(PropTypes.oneOfType([optionPropType, optGroupPropType])),
+  /**
+   * Placeholder data. The presence of this data object is used as the indicator for whether or not a placeholder is used.
+   */
+  placeholder: PropTypes.shape({
+    /**
+     * Whether the placeholder is included as an option to clear selection.
+     */
+    allowClear: PropTypes.bool,
+    /**
+     * Optional override of the default display.
+     */
+    display: PropTypes.string,
+    /**
+     * Optional override of the default value.
+     */
+    value: PropTypes.string,
+  }),
+  /**
+   * Whether the field is required.
+   */
+  required: PropTypes.bool,
+  /**
+   * The value of the select. Can be a string or number.
+   */
+  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+};
+
+const defaultProps = {
+  disabled: false,
+  isIncomplete: false,
+  isInvalid: false,
+  options: [],
+  required: false,
+};
+
+const createPlaceholder = (placeholder, intl) => {
+  if (!placeholder) {
+    return undefined;
+  }
+  const display = placeholder.display ? placeholder.display : intl.formatMessage({ id: 'Terra.form.select.defaultDisplay' });
+  const value = placeholder.value ? placeholder.value : defaultPlaceholderValue;
+  const attrs = placeholder.allowClear ? {} : { disabled: true, hidden: true };
+  return <option value={value} {...attrs}>{display}</option>;
+};
+
+const createOptions = options => {
+  const currentOptGroupKeys = [];
+
+  return options.map(current => {
+    if (current.options) {
+      const optGroupKeyIndex = geOptGroupKeyIndex(current.display, currentOptGroupKeys);
+      const optGroupKey = getOptGroupKey(current.display, optGroupKeyIndex);
+      currentOptGroupKeys.push(optGroupKey);
+
+      return (
+        <optgroup key={optGroupKey} label={current.display}>
+          {createOptions(current.options)}
+        </optgroup>
+      );
+    }
+    return <option key={`${current.value}`} value={current.value}>{current.display}</option>;
+  });
+};
+
+const NativeSelect = ({
+  ariaDescribedBy,
+  ariaLabel,
+  attrs,
+  disabled,
+  defaultValue,
+  id,
+  intl,
+  isInvalid,
+  isIncomplete,
+  onChange,
+  options,
+  placeholder,
+  required,
+  value,
+  ...customProps
+}) => {
+  const [uncontrolledValue, setUncontrolledValue] = useState(!value ? defaultValue || getFirstValue(options, placeholder) : undefined);
```

Comment:
The evaluation of the `value` prop may need to account for an empty string, in this case an empty string would always flip the select into an uncontrolled. Generally empty string is the placeholder/cleared value representation.

Reply:
!"" evaluates to true, so any value passed on the initial mount should satisfy the condition. Also, it's in progress, but the placeholder is being removed as well.

Reply:
It looks like an empty string will set the uncontrolledValue as the first option in the list using `getFirstValue`. I think this will make `uncontrolledValue ` and `refIsControlled` ambiguous. It's a little goofy, I've seen a few teams in the past using an empty string as a value

```jsx
// (depending on if the empty string option is the first in the list)
// This will initialize `uncontrolledValue` as empty string and `refIsControlled` as true
<NativeSelect value='' />
```

```jsx
// (depending on if the empty string option is the first in the list)
// This will initialize `uncontrolledValue` as empty string and `refIsControlled` as true
<NativeSelect defaultValue='' />
```

Reply:
Updated with:
```
  const refIsControlled = useRef(!!value);
```

<NativeSelect defaultValue='' />
// This results in:
// uncontrolledValue: ""
// refIsControlled: false

Reply:
I see the issue now, my bad.

Reply:
Updated.